Fix incorrect handling of blobless blocks

henridf opened the pull request:
We were passing all blocks to local processing, including those for which we were missing blobs. This commit fixes that.

tersec commented on beacon_chain/gossip_processing/eth2_processor.nim:
```diff
@@ -242,6 +242,13 @@ proc processSignedBeaconBlock*(
       if self.blobQuarantine[].hasBlobs(signedBlock):
         blobs = self.blobQuarantine[].popBlobs(signedBlock.root)
+
+        self.blockProcessor[].addBlock(
```
This appears to only call `self.blockProcessor[].addBlock(...)` at all `when typeof(signedBlock).toFork() >= ConsensusFork.Deneb`?

henridf replied:
indeed, some crossed wires there 🤪 ... should be fixed.

henridf committed:
Fix incorrect handling of blobless blocks

We were passing along all blocks to local processing, including those
for which we were missing blobs. This commit fixes that.
